### ISSUE TITLE
Add checkstyle compatible reporter

### ISSUE THREAD

**Issue by christofdamian** [sds/slim-lint#34]
This would make it much easier to integrate with Jenkins. 

Rubocop for example has a plugin to do this: https://github.com/eitoball/rubocop-checkstyle_formatter

The format itself is pretty simple. For example:
```xml
<?xml version='1.0'?>
<checkstyle>
  <file name='Gemfile'/>
  <file name='config.ru'>
    <error line='37' column='27' severity='info' message='Avoid using `{...}` for multi-line blocks.' source='com.puppycrawl.tools.checkstyle.Style/BlockDelimiters'/>
    <error line='39' column='20' severity='info' message='Redundant curly braces around a hash parameter.' source='com.puppycrawl.tools.checkstyle.Style/BracesAroundHashParameters'/>
  </file>
```

**Comment by christofdamian** [comment]
I am going to give implementing it a try too :-)

**Comment by sds** [comment]
Implemented in #35. Thanks!
